src/CorrugatedIron.Tests.Live/MapReduce/WhenUsingSearchAsInput.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2011 - OJ Reeves & Jeremiah Peschka
+//
+// This file is provided to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License.  You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Linq;
+using CorrugatedIron.Comms;
+using CorrugatedIron.Extensions;
+using CorrugatedIron.Models;
+using CorrugatedIron.Models.MapReduce;
+using CorrugatedIron.Models.MapReduce.Inputs;
+using CorrugatedIron.Models.Search;
+using CorrugatedIron.Tests.Extensions;
+using CorrugatedIron.Tests.Live.Extensions;
+using CorrugatedIron.Util;
+using NUnit.Framework;
+
+namespace CorrugatedIron.Tests.Live.MapReduce
+{
+    [TestFixture]
+    public class WhenUsingSearchAsInput : RiakMapReduceTestBase
+    {
+        private const string BucketType = "search_type";
+        private new const string Bucket = "yoko_bucket";
+        private const string Index = "yoko_index";
+        private const string RiakSearchKey = "a.hacker";
+        private const string RiakSearchKey2 = "a.public";
+        private readonly Random _random = new Random();
+        private int _randomId;
+        private RiakObjectId _alyssaRiakId;
+
+        // See https://raw.githubusercontent.com/basho/yokozuna/develop/priv/default_schema.xml for dynamic field suffix meanings.
+        private const string RiakSearchDoc =
```

Comment:
Could these be made `@""` strings? At least then they could be broken across lines.